pkg/autoscaler/stats_scraper.go

```diff
@@ -17,29 +17,37 @@ limitations under the License.
 package autoscaler
 
 import (
+	"context"
 	"errors"
 	"fmt"
 	"io"
 	"net/http"
 	"time"
 
+	"github.com/knative/pkg/logging"
 	"github.com/knative/serving/pkg/apis/serving"
 	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
 	"github.com/knative/serving/pkg/reconciler"
 	dto "github.com/prometheus/client_model/go"
 	"github.com/prometheus/common/expfmt"
 	"go.uber.org/zap"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
 )
 
 const (
 	httpClientTimeout = 3 * time.Second
+
+	cacheTimeout = time.Second
+
+	scraperPodName = "service-scraper"
```

Comment:
Please add a comment why we're doing this.

Reply:
Done.